### PR TITLE
analyze: fix offline checks for syscall filter and 'native' architecture

### PR DESCRIPTION
**Pull request by bluca**
Fixes https://github.com/systemd/systemd/issues/22914

### CODE REVIEW

**Comment by yuwata** [comment]
```
2022-04-01T01:15:13.5825842Z ../src/analyze/analyze-security.c: In function ‘assess_system_call_architectures’:
2022-04-01T01:15:13.5826928Z ../src/analyze/analyze-security.c:554:87: error: ‘native’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
2022-04-01T01:15:13.5827641Z   554 |         } else if (set_contains(info->system_call_architectures, UINT32_TO_PTR(native + 1)) &&
2022-04-01T01:15:13.5828032Z       |                                                                                       ^
2022-04-01T01:15:13.5828653Z cc1: all warnings being treated as errors
```

File: src/analyze/analyze-security.c [hunk 1]
```diff
@@ -530,23 +530,28 @@ static int assess_restrict_namespaces(
...
                 char **ret_description) {
 
+        uint32_t native;
         char *d;
         uint64_t b;
 
         assert(ret_badness);
         assert(ret_description);
 
+        assert(seccomp_arch_from_string("native", &native) == 0);
```
Comment (yuwata) [inline src/analyze/analyze-security.c]: assert -> assert_se

Also, I'd like the condition `== 0` -> `>= 0`.

**Review by yuwata — APPROVED** [review]
LGTM.

**Comment by poettering** [comment]
lgtm

**Comment by yuwata** [comment]
The CentOS CI failure is unrelated. The test is theoretically racy.
```
19:04:10 /* test_sd_device_enumerator_filter_subsystem */
19:04:10 Assertion 'hashmap_isempty(h)' failed at src/libsystemd/sd-device/test-sd-device.c:196, function unsigned int test_sd_device_enumerator_filter_subsystem_one(const char *, Hashmap *)(). Aborting.
```

**Pull request merged**